src/node/internal/internal_http_incoming.ts

```diff
@@ -67,6 +76,50 @@ export class IncomingMessage extends Readable implements _IncomingMessage {
     ): void => {
       incoming.#setFetchResponse(response);
     };
+
+    // This method sets the socket property of the IncomingMessage object.
+    // Please rest assured that this method implements a subset of Socket since
+    // in Node.js it's net.Socket which isn't possible to implement within our own
+    // implementation since our implementation is based on Request and Response objects.
+    setIncomingMessageSocket = (
+      incoming: IncomingMessage,
+      { headers, localPort }: { headers: Headers; localPort: number }
+    ): void => {
+      const connectingIp = headers.get('cf-connecting-ip');
+      const isConnectingIpIpv4 = connectingIp ? isIPv4(connectingIp) : true;
+      const remotePort = Math.floor(Math.random() * (65535 - 32768 + 1));
+
+      incoming.#socket = {
+        encrypted: headers.get('x-forwarded-proto') === 'https',
```

Comment:
Should this be a readonly-getter?

Reply:
It is not a read-only property. Ref: https://github.com/nodejs/node/blob/bfffd44457ed10d93a450660df170662aa57e9ad/lib/internal/tls/wrap.js#L565

Reply:
Sigh. silly node.js

Reply:
Ping @anonrig